feature-libs/cart/quick-order/core/services/quick-order.service.ts

```diff
@@ -16,24 +17,38 @@ import {
   ProductSearchPage,
   SearchConfig,
 } from '@spartacus/core';
-import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import {
+  BehaviorSubject,
+  Observable,
+  Subject,
+  Subscription,
+  timer,
+} from 'rxjs';
 import { filter, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { ClearMessageTimouts } from '../models/clear-message-timeouts.model';
 
-@Injectable({
-  providedIn: 'root',
-})
-export class QuickOrderService implements QuickOrderFacade {
+@Injectable()
+export class QuickOrderService implements QuickOrderFacade, OnDestroy {
   protected productAdded$: Subject<string> = new Subject<string>();
   protected entries$: BehaviorSubject<OrderEntry[]> = new BehaviorSubject<
     OrderEntry[]
   >([]);
+  protected deletedEntries$: BehaviorSubject<DeletedEntriesObject> =
+    new BehaviorSubject<DeletedEntriesObject>({});
+  protected deletionClearTimeout = 10000;
+
+  private clearMessageTimeouts: ClearMessageTimouts = {};
```

Comment:
I don't think this should have in name `message`. 

Service should not be aware of UI.